.github/workflows/codecov-main.yaml

```diff
@@ -0,0 +1,19 @@
+name: Codecov report on push to the "main" branch
```

Comment:
codecov is used in many [openshift org's repositories](https://app.codecov.io/gh/openshift) and also in [redhat-appstudio's](https://app.codecov.io/gh/redhat-appstudio).

I do not know whether there's a strict requirement to use only this tool.
However the codecov suggestions/warning do not force developers to address all of them, it's up to an individual person to decide whether it makes sense to take some actions on them.

The main purpose of using this tool is to provide an information about a test coverage and suggest how it can be improved. Also there are ways to configure this tool, i.e. possibly we might be able to remove the codecov code suggestions in case they prove to be too much annoying :)

Reply:
golanci-lint - I will try to address it in this PR